packages/editor/src/ui-tools/viewport/ViewportBar.tsx

```diff
@@ -0,0 +1,24 @@
+import { ActionButton, type ActionTrigger } from '@axonivy/process-editor-view';
+import { Action } from '@eclipse-glsp/client';
+import React from 'react';
+
+type ViewportBarButton = ActionTrigger;
+
+type ViewportBarProps = {
+  triggers: ViewportBarButton[];
+  onAction: (action: Action) => void;
+  zoomLevel: string;
+};
+
+const ViewportBar = React.memo(({ triggers, onAction, zoomLevel }: ViewportBarProps) => (
+  <div className='viewport-bar'>
+    <div className='viewport-bar-tools'>
+      {triggers.map(trigger => (
+        <ActionButton key={trigger.id} trigger={trigger} onAction={onAction} />
+      ))}
```

Comment:
I think I would directly add these here as children.

Reply:
Yeah, I agree, it is much nicer that way!